src/pagination/select.js

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import * as React from 'react';
+import {Select as DefaultSelect, TYPE} from '../select';
+import {StyledListItem} from '../menu';
+import {SelectInput, SelectTag} from './styled-components';
+import type {SelectMenuItemT} from './types';
+
+const noop = () => {};
+
+const getOptionLabel = item => item.label;
+
+// This is used internally only, not meant to be exported
+
+export default function Select({
+  currentPage,
+  options,
+  ...restProps
+}: {
+  currentPage: number,
+  options: Array<SelectMenuItemT>,
+}) {
+  return (
+    <DefaultSelect
+      // Required to re-initialize Select component so that selectedOptions work
+      key={currentPage}
+      options={options}
+      selectedOptions={[options[currentPage - 1]]}
+      rows={5}
+      type={TYPE.select}
+      getOptionLabel={getOptionLabel}
+      overrides={{
+        Input: SelectInput,
+        Tag: SelectTag,
+        DropDownItem: StyledListItem,
+      }}
+      // None of these props should be required
+      filterOption={() => false}
+      filterable={false}
+      multiple={false}
+      error={false}
+      textValue={''}
+      onBlur={noop}
+      onFocus={noop}
+      onMouseEnter={noop}
+      onMouseLeave={noop}
+      onMouseDown={noop}
+      onMouseUp={noop}
```

Comment:
I think flow is smart enough to know that a prop is not required if a defaultProp is set for it, even if PropsT has marked it as required ([reference](https://flow.org/en/docs/react/components/#toc-using-default-props)). It looks like Select sets defaultProps for most of these, were you still seeing flow errors when omitting them?

Reply:
Yep. Maybe it's just with my particular installation? I think some consolidation is needed here regardless. We should just *not* type `defaultProps` if it's causing us so much headache (we're already skipping that step in other components as well).

Reply:
Expanding on that, I mean removing the typedef here: https://github.com/uber-web/baseui/blob/master/src/select/select.js#L33 and properly mark `PropsT` with what's required/what's not.

Reply:
Yeah that probably makes sense.

Reply:
You were right, flow recognizes `defaultProps` properly. The issue here was that we were typing `defaultProps` explicitly to something else. Removing that typedef for `defaultProps` produced the expected behavior.